library/src/main/java/com/getstream/sdk/chat/navigation/ChatNavigatorImpl.kt

```diff
@@ -0,0 +1,22 @@
+package com.getstream.sdk.chat.navigation
+
+import com.getstream.sdk.chat.navigation.destinations.ChatDestination
+
+class ChatNavigatorImpl : ChatNavigator {
+    private var handler: ChatNavigationHandler? = null
+
+    override fun setHandler(handler: ChatNavigationHandler) {
+        this.handler = handler
+    }
+
+    override fun navigate(destination: ChatDestination) {
+        val handler = handler
```

Comment:
Why do we need this local variable?

Reply:
`handler` is a mutable var, so smart casts don't work on it otherwise. The simplest the checks here get is what you see on the next line. The temporary variable could be introduced with a `let {}`, but it's really not any more readable.

Reply:
What about having a "Non ChatNavitationHandler" that return false and avoid null values?

```
private var handler: ChatNavigationHandler = object  : ChatNavigationHandler {
    override fun navigate(destination: ChatDestination?): Boolean = false
}
```

Reply:
Ah, I thought I allowed setting `null` handlers through the setter function, but apparently not. If we don't need to be able to `null` out the `handler` later on, this is a great idea, I'll add it.